packages/block-editor/src/store/selectors.js

```diff
@@ -1681,18 +1683,11 @@ export function canInsertBlocks( state, clientIds, rootClientId = null ) {
  * @return {boolean} Whether the given block is allowed to be removed.
  */
 export function canRemoveBlock( state, clientId, rootClientId = null ) {
-	const attributes = getBlockAttributes( state, clientId );
-	if ( attributes === null ) {
-		return true;
-	}
-	if ( attributes.lock?.remove !== undefined ) {
-		return ! attributes.lock.remove;
-	}
-	if ( getTemplateLock( state, rootClientId ) ) {
-		return false;
-	}
-
-	return getBlockEditingMode( state, rootClientId ) !== 'disabled';
+	return canRemoveBlockCheck(
+		getBlockAttributes( state, clientId ),
+		getBlockEditingMode( state, rootClientId ),
+		getTemplateLock( state, rootClientId )
+	);
```

Comment:
It's ok to get all these at once here: most blocks have attributes and have no parent template lock, so the previous early returns have no effect.